data_model/src/lib.rs

```diff
@@ -1248,46 +1139,46 @@ pub mod asset {
     }
 
     macro_rules! impl_try_as_for_asset_value {
-        ( $($variant:ident( $ty:ty ),)* ) => {$(
-            impl TryAsMut<$ty> for AssetValue {
-                type Error = Error;
-
-                fn try_as_mut(&mut self) -> Result<&mut $ty> {
-                    if let AssetValue:: $variant (value) = self {
-                        Ok(value)
-                    } else {
-                        Err(eyre!(
-                            concat!(
-                                "Expected source asset with value type:",
-                                stringify!($variant),
-                                ". Got: {:?}",
-                            ),
-                            self.value_type()
-                        ))
-                    }
-                }
-            }
-
-            impl TryAsRef<$ty> for AssetValue {
-                type Error = Error;
-
-                fn try_as_ref(&self) -> Result<& $ty > {
-                    if let AssetValue:: $variant (value) = self {
-                        Ok(value)
-                    } else {
-                        Err(eyre!(
-                            concat!(
-                                "Expected source asset with value type:",
-                                stringify!($variant),
-                                ". Got: {:?}",
-                            ),
-                            self.value_type()
-                        ))
-                    }
-                }
-            }
-        )*}
-    }
+		( $($variant:ident( $ty:ty ),)* ) => {$(
+			impl TryAsMut<$ty> for AssetValue {
+				type Error = Error;
+
+				fn try_as_mut(&mut self) -> Result<&mut $ty> {
+					if let AssetValue:: $variant (value) = self {
+						Ok(value)
+					} else {
+						Err(eyre!(
+							concat!(
+								"Expected source asset with value type:",
+								stringify!($variant),
+								". Got: {:?}",
+							),
+							self.value_type()
+						))
+					}
+				}
+			}
+
+			impl TryAsRef<$ty> for AssetValue {
+				type Error = Error;
+
+				fn try_as_ref(&self) -> Result<& $ty > {
+					if let AssetValue:: $variant (value) = self {
+						Ok(value)
+					} else {
+						Err(eyre!(
+							concat!(
+								"Expected source asset with value type:",
+								stringify!($variant),
+								". Got: {:?}",
+							),
+							self.value_type()
+						))
+					}
+				}
+			}
+		)*}
+	}
```

Comment:
how come it sometimes happens that spaces get replaced with tabs in your code?

Reply:
`emacs`'s aggressive indent has a habit of doing that. I thought I fixed it during last PR, apparently not.